src/checkers/inference/VariableAnnotator.java

```diff
@@ -1175,12 +1175,15 @@ private void shallowAnnotateArray(AnnotatedTypeMirror type, Tree tree, int level
             TreePath pathToTopLevelTree = inferenceTypeFactory.getPath(topLevelTree);
 
             AnnotationLocation location;
-            // 'pathToTopLevelTree' is null when it's an artificial array creation tree like for
-            // varargs. We don't need to create AstPathLocation for them.
-            if (pathToTopLevelTree != null) {
-                ASTRecord astRecord = ASTPathUtil.getASTRecordForPath(inferenceTypeFactory, pathToTopLevelTree).newArrayLevel(level);
+
+            assert pathToTopLevelTree != null;
+            ASTRecord astRecord = ASTPathUtil.getASTRecordForPath(inferenceTypeFactory, pathToTopLevelTree);
+            if (astRecord != null) {
+                astRecord = astRecord.newArrayLevel(level);
                 location = new AstPathLocation(astRecord);
             } else {
+                // astRecord for `topLevelTree` is null when it's an artificial array creation tree like for
```

Comment:
Does this branch actually get triggered? It's a bit odd that the comment moved here.
Can you add a test case that triggers this and use isHackMode to guard this work-around?

Reply:
The else branch is triggered when calling `ASTPathUtil.getASTRecordForPath` on an artificial tree. An example is methods with varargs.
```
void foo(Object o1, Object o2) {
        bar();
    }

    void bar(Object... args) {
        
    }
```

This seems more of a bug than an issue that needs to be hacked. And every other use of `ASTPathUtil.getASTRecordForPath` check if null is returned.